src/Extensions/ElementalAreaUsedOnTableExtension.php

```diff
@@ -3,24 +3,80 @@
 namespace DNADesign\Elemental\Extensions;
 
 use SilverStripe\Admin\Forms\UsedOnTable;
+use SilverStripe\CMS\Model\SiteTree;
 use SilverStripe\ORM\DataExtension;
 use SilverStripe\ORM\ArrayList;
 use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\ValidationException;
 use DNADesign\Elemental\Models\ElementalArea;
+use DNADesign\Elemental\Models\BaseElement;
 
 class ElementalAreaUsedOnTableExtension extends DataExtension
 {
 
     /**
      * Hides ElementalArea's from the "Used On" tab when viewing files
      *
+     * @var array $excludedClasses
+     */
+    public function updateUsageExcludedClasses(array &$excludedClasses)
+    {
+        $excludedClasses[] = ElementalArea::class;
+    }
+
+    /**
+     * Legacy function kept for semver, replaced with updateUsageExcludedClasses above
+     *
      * @return void
-     * @var ArrayList $dataObjects
+     * @var ArrayList $usage
      * @var DataObject $record
      * @see UsedOnTable::updateUsage
+     * @deprecated 4.5.0 Use self::updateUsageExcludedClasses instead
+     */
+    public function updateUsage(ArrayList &$usage, DataObject &$record)
+    {
+        // noop
+    }
+
+    /**
+     * Exclude content blocks that aren't linked to a page
+     *
+     * @param bool $excludeDataObject
+     * @param DataObject $dataObject
+     */
+    public function updateUsageDataObject(DataObject $dataObject)
+    {
+        if (!($dataObject instanceof BaseElement)) {
+            return;
+        }
+        try {
+            if (!$dataObject->getPage()) {
+                $dataObject = null;
+            }
+        } catch (ValidationException $e) {
```

Comment:
I _think_ this error boundary is unnecessary - `BaseElement::getPage()` shouldn't throw a `ValidationException`, despite its docblock stating otherwise.

Reply:
It might be, though that doesn't stop phpstorm from complaining